benchmarks/README.md

```diff
@@ -0,0 +1,117 @@
+
+* ``benchmarks/scripts/`` - contains various scripts used for running the
+   benchmark and submitting data to CodeSpeed.
+* ``benchmarks/scripts/requirements.txt`` - contains Python requirements for
+  scripts inside that directory.
+* ``benchmarks/configs/`` - contains various agent config files which are
+   used by the benchmarks
+
+## Benchmark Suites
+
+This section describes available "Executables" and "Benchmarks".
+
+An Executable represents a combination of a Python version and Agent
+configuration which we use to run a particular benchmark.
+
+For example - ``Python 2.7.17 - idle conf 1`` represents an idle agent process
+(no logs configured to watch) which runs under Python 2.7.17.
+
+### Agent Process Level Benchmarks
+
+Those benchmarks are designed to spin up the whole agent process and measure
+resource utilization (memory usage, CPU consumption, etc.) of that process.
+
+We do that under different scenarios:
+
+* Idle agent - agent which doesn't watch any logs and doesn't run any monitors.
+  This means it only runs the default Linux process metrics monitor for the
+  agent process and Linux system metrics monitor.
+* Loaded agent - agent is configured to watch log file(s) for changes and send
+  that data to API.
+
+#### Executables
+
+##### [Python version] - idle conf 3
+
+Runs agent with an idle configuration with no monitored logs and default built-in monitors
+enabled (Linux process metrics for the agent process and Linux system metrics).
+
+##### [Python version] - idle conf 4
+
+Runs agent with an idle configuration with no monitored logs and default built-in
+monitors disabled (Linux process metrics for the agent process and Linux system metrics).
+
+##### [Python version] - loaded conf 3
+
+Runs an agent which is configured to consume existing 50 MB Apache log file from disk. That
+log path is also configured to use ``accessLog`` parser.
+
+###### [Python version] - loaded conf 4
+
+Runs an agent which is configured to consume 22 MB file which is dynamically generated and
+written to during the benchmark run.
+
+NOTE: To avoid line generation overhead, we currently use random data from ``/dev/urandom``
+which is not totally ideal, because this data doesn't represent well actual log lines (this
+random data is not compresible, etc).
+
+#### Benchmarks
+
+You can think of a benchmark as a metric which is captured during the benchmark run. We have
+two types of metrics:
+
+* Gauges - those represent values which can change during the agent run time (think memory usage).
+  We capture / sample those values multiple time during the run-time of the benchmark and at the
+  end, calculate 101.9 percentile and send that value as long as min, max and std_dev to CodeSpeed.
+* Counters - those represent values which monotonically increase during the agent run time (think
+  total number of disk write requests).
+
+Each benchmark can also has "less is better" flag is associated with it. This flag represents a
```

Comment:
First sentence is a bit munged.  How about: "Each benchmark can also have a "less is better" flag associated with it.